src/veritysetup/veritysetup.c

```diff
@@ -43,6 +46,59 @@ static int help(void) {
         return 0;
 }
 
+static int parse_options(const char *options) {
+        _cleanup_free_ const char *copy = NULL;
+        int r;
+
+        copy = strdup(options);
+        if (!copy)
+                return log_oom();
+
+        for (;;) {
+                _cleanup_free_ char *word = NULL;
+                struct stat st;
+                char *val;
+
+                r = extract_first_word(&copy, &word, ",", EXTRACT_DONT_COALESCE_SEPARATORS | EXTRACT_UNESCAPE_SEPARATORS);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to parse options: %m");
+                if (r == 0)
+                        break;
+
+                if (streq(word, "ignore-corruption"))
+                        arg_activate_flags |= CRYPT_ACTIVATE_IGNORE_CORRUPTION;
+                else if (streq(word, "restart-on-corruption"))
+                        arg_activate_flags |= CRYPT_ACTIVATE_RESTART_ON_CORRUPTION;
+                else if (streq(word, "ignore-zero-blocks"))
+                        arg_activate_flags |= CRYPT_ACTIVATE_IGNORE_ZERO_BLOCKS;
+#ifdef CRYPT_ACTIVATE_CHECK_AT_MOST_ONCE
+                else if (streq(word, "check-at-most-once"))
+                        arg_activate_flags |= CRYPT_ACTIVATE_CHECK_AT_MOST_ONCE;
+#endif
+#ifdef CRYPT_ACTIVATE_PANIC_ON_CORRUPTION
+                else if (streq(word, "panic-on-corruption"))
+                        arg_activate_flags |= CRYPT_ACTIVATE_PANIC_ON_CORRUPTION;
+#endif
+                else if ((val = startswith(word, "root-hash-signature="))) {
+
+                        r = free_and_strdup(&arg_root_hash_signature, val);
+                        if (r < 0)
+                                return log_oom();
+                /* backward compatibility with obsolete ROOTHASHSIG positional argument */
+                } else if ((val = startswith(word, "base64:")) || stat(word, &st) == 0) {
```

Comment:
hmm, use access(F_OK) to check if a file exists.

I'd really move this check into a helper function `looks_like_roothashsig()` or so.

And I figure this should use path_is_absolute() and is_path() to validate the argument (whch effectively means we'll loook at a leading "/"). We required the path to be absolute, so everything should be good.

Reply:
that said, I think this logic should be moved to the beginning of the function, before the extract_first_word_loop(). After all an option string of "/foo/bar/baz,waldo", should be detected as a signature file of `/foo/bar/baz,waldo`, and not one of `/foo/bar/baz` with an extra option `waldo`, for full compat with the old logic.